zipline/data/data_portal.py

```diff
@@ -154,7 +154,11 @@ def __init__(self,
                  minute_history_prefetch_length=_DEF_M_HIST_PREFETCH,
                  daily_history_prefetch_length=_DEF_D_HIST_PREFETCH):
 
-        self.trading_calendar = trading_calendar
+        if trading_calendar is not None:
```

Comment:
It looks like we do this check in `run_algo._run` [as well](https://github.com/quantopian/zipline/pull/2026/files#diff-ec1e1e05a956fe81da0703490710af21L159). Since that's the only call site that now might pass a `None` trading_calendar, can we do the check just there instead of here too?